### PR TITLE
feat: Improve when tile layers are idle.

### PR DESCRIPTION
**Pull request by manthey**
Add a setting to allow tile layers to be considered idle either when all tiles are finished loading or as soon as all tiles in the view are loaded.  Before this was always equivalent to all tiles.  The new default is just tiles in the view, as this is more relevant to typical use.

feat: Allow removing promises from the idle monitor.

This will facilitate tile layers reporting that they are idle when all tiles in the view are loaded even if tiles out of the view are still loading.

### CODE REVIEW

**Comment by manthey** [comment]
@aashish24 This continues improvements that will build to an explicit A/B buffered tile layer.

File: src/tileLayer.js [hunk 1]
```diff
@@ -1358,6 +1366,17 @@ var tileLayer = function (arg) {
           m_this._purge(zoom, true);
         }
       );
+    // for tiles that aren't in view, remove them from the list of tiles that
+    // are needed to be loaded to be considered idle.
+    if (m_this._options.idleAfter !== 'all') {
+      for (const hash in m_promisedTiles) {
+        const tile = m_promisedTiles[hash];
+        if (tiles.indexOf(tile) < 0) {
```
Comment (aashish24) [inline src/tileLayer.js:1374]: So is that the case that all the tiles that are not in view have index < 0, just wanted to make sure I understand the logic here.

Comment (manthey) [reply src/tileLayer.js:1374]: Yes.  If a value is not in an array, indexOf returns -1.  The tiles array is the list of tiles in the view.

Comment (aashish24) [reply src/tileLayer.js:1374]: Okay thanks.

**Review by aashish24 — APPROVED** [review]

**Pull request merged**